x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/PartiallyCachedShardAllocationIntegTests.java

```diff
@@ -103,11 +104,22 @@ private MountSearchableSnapshotRequest prepareMountRequest() throws InterruptedE
         );
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/93868")
     public void testPartialSearchableSnapshotNotAllocatedToNodesWithoutCache() throws Exception {
         final MountSearchableSnapshotRequest req = prepareMountRequest();
         final RestoreSnapshotResponse restoreSnapshotResponse = client().execute(MountSearchableSnapshotAction.INSTANCE, req).get();
-        assertThat(restoreSnapshotResponse.getRestoreInfo().successfulShards(), equalTo(0));
+        try {
+            assertThat(restoreSnapshotResponse.getRestoreInfo().successfulShards(), equalTo(0));
+        } catch (NullPointerException | AssertionError e) {
+            var explain = client().admin()
+                .cluster()
+                .prepareAllocationExplain()
+                .setIndex(req.mountedIndexName())
+                .setShard(0)
+                .setPrimary(true)
+                .get();
+            logger.info("Failed to allocate: {}", Strings.toString(explain.getExplanation()));
+            throw e;
+        }
```

Comment:
I executed this test ~500 times locally but could not reproduce.
All the recorded failures has happened either because `restoreSnapshotResponse.getRestoreInfo()` is null meaning restore have not happened or because it timed out. I am hoping this could provide additional information about the cause of the failure.